### PR TITLE
[CARBONDATA-4119][CARBONDATA-4238][CARBONDATA-4237][CARBONDATA-4236] Support geo insert without geoId and document changes

### PR DESCRIPTION
**Pull request by ShreelekhyaG**
### Why is this PR needed?
 - To insert without geoid (like load) on geo table.
 - [CARBONDATA-4119] : User Input for GeoID column not validated.
 - [CARBONDATA-4238] : Documentation Issue in ddl-of-carbondata.md#add-columns
 - [CARBONDATA-4237] : Documentation issues in streaming-guide.md, file-structure-of-carbondata.md and sdk-guide.md.
 - [CARBONDATA-4236] : Documenatation issues in configuration-parameters.md.
 - import processing class in streaming-guide.md is wrong
 
 ### What changes were proposed in this PR?
- Made changes to support insert on geo table with auto-generated geoId.
-  [CARBONDATA-4119] : Added documentation about insert with custom geoId. Changes in docs/spatial-index-guide.md
- Other documentation changes added.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 3.1, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_3.1/264/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/4119/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5861/

**Comment by Indhumathi27** [comment]
@ShreelekhyaG can you update PR description with JIRA and that related code change (file name)

File: docs/spatial-index-guide.md [hunk 1]
```diff
@@ -91,6 +91,15 @@ Note:
 | SPATIAL_INDEX.xxx.conversionRatio | Conversion factor. It allows user to translate longitude and latitude to long. For example, if the data to load is longitude = 13.123456, latitude = 101.12356. User can configure conversion ratio sub-property value as 1000000, and change data to load as longitude = 13123456 and latitude = 10112356. Operations on long is much faster compared to floating-point numbers.|
 | SPATIAL_INDEX.xxx.class | Optional user custom implementation class. Value is fully qualified class name.|
 
+### Load/Insert
+Load/Insert with no geoId column, then geoId will be generated internally.
+```
+insert into source_index select 1,116.285807,40.084087;
+```
+Insert with custom geoId
```
Comment (Indhumathi27) [inline docs/spatial-index-guide.md:99]: can mention load with custom geo Id is supported or not

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 3.1, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_3.1/276/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5873/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/4131/

File: docs/spatial-index-guide.md (hunk 1 above)
Comment (ShreelekhyaG) [reply docs/spatial-index-guide.md]: Done

File: docs/spatial-index-guide.md [hunk 2]
```diff
@@ -91,6 +91,15 @@ Note:
 | SPATIAL_INDEX.xxx.conversionRatio | Conversion factor. It allows user to translate longitude and latitude to long. For example, if the data to load is longitude = 13.123456, latitude = 101.12356. User can configure conversion ratio sub-property value as 1000000, and change data to load as longitude = 13123456 and latitude = 10112356. Operations on long is much faster compared to floating-point numbers.|
 | SPATIAL_INDEX.xxx.class | Optional user custom implementation class. Value is fully qualified class name.|
 
+### Load/Insert
+Load/Insert with no geoId column, then geoId will be generated internally.
+```
+insert into source_index select 1,116.285807,40.084087;
+```
+Load/Insert with custom geoId
```
Comment (Indhumathi27) [inline docs/spatial-index-guide.md:99]: can you add testcase for load also

Comment (ShreelekhyaG) [reply docs/spatial-index-guide.md:99]: Already test case exists for load - `test("test load data with customized correct spatial index to geo table and polygon query")`

**Comment by ShreelekhyaG** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 3.1, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_3.1/282/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5880/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/4137/

**Comment by Indhumathi27** [comment]
LGTM

**Pull request closed without merging**